doc/sphinx/proof-engine/tactics.rst

```diff
@@ -1257,115 +1257,82 @@ Managing the local context
    for any unresolved variables rather than failing.  Typically this happens when
    using a ``%`` intropattern (see :n:`@simple_intropattern`).
 
-.. tacn:: clear @ident
-   :name: clear
+.. tacn:: clear {| {? {? - } {+ @ident } } | @natural }
 
-   This tactic erases the hypothesis named :n:`@ident` in the local context of
-   the current goal. As a consequence, :n:`@ident` is no more displayed and no
-   more usable in the proof development.
+   Erases hypotheses from the context of the current goal, provided that they are
+   not referenced directly or indirectly from the remaining hypotheses or the goal.
+   That means the hypotheses will no longer be shown and can no longer be
+   used in the proof.
 
-   .. exn:: No such hypothesis.
-      :undocumented:
+   If no arguments are given, all hypotheses are erased except for the ones the
+   goal depends on.  If the goal depends on all hypotheses, the context is left
+   unchanged.  Otherwise, the specified hypotheses and all hypotheses that
+   depend on them that the goal doesn't depend on are erased.
+   Specifying `-` erases all hypotheses except those listed.
+
+   .. todo arguably, "clear -" should be the syntax for removing all unreferenced goals
+      instead of "clear".
 
    .. exn:: @ident is used in the conclusion.
       :undocumented:
 
    .. exn:: @ident is used in the hypothesis @ident.
       :undocumented:
 
-   .. tacv:: clear {+ @ident}
-
-      This is equivalent to :n:`clear @ident. ... clear @ident.`
-
-   .. tacv:: clear - {+ @ident}
-
-      This variant clears all the hypotheses except the ones depending in the
-      hypotheses named :n:`{+ @ident}` and in the goal.
-
-   .. tacv:: clear
-
-      This variants clears all the hypotheses except the ones the goal depends on.
+   .. tacn:: clearbody {+ @ident }
 
-   .. tacv:: clear dependent @ident
-
-      This clears the hypothesis :token:`ident` and all the hypotheses that
-      depend on it.
-
-   .. tacv:: clearbody {+ @ident}
-      :name: clearbody
-
-      This tactic expects :n:`{+ @ident}` to be local definitions and clears
-      their respective bodies.
+      This tactic expects :n:`{+ @ident}` to be :term:`local definitions <context-local definition>`
+      and clears their respective bodies.
       In other words, it turns the given definitions into assumptions.
 
       .. exn:: @ident is not a local definition.
          :undocumented:
```

Comment:
Is this really undocumented if it is self-explanatory?

Reply:
That's a Sphinx directive.  It's not seem by the user. If there's no paragraph after the exn Sphinx reports an error.